Add screencast functionality

bradleyscollins opened the pull request:
This pull request adds an API that exposes CDP's screencast capability. 

See the following:

- [Page.startScreencast](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-startScreencast)
- [Page.stopScreencast](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-stopScreencast)

**Note**: At this time, the Chrome DevTools Protocol documentation still has screencast methods/types marked as "experimental."

bradleyscollins commented:
A number of the test failures are on account of a difference between expected vs. actual screenshot height.

We noticed that as of Chrome 128, the heights of our screenshot/screencast images were different from (less than) the height specified in the `window_size` option of the `Browser` constructor. We had to set the viewport size explicitly to get the expected screenshot height.

route commented:
@bradleyscollins could you please rebase on master? after that we can merge, should be fixed in there

bradleyscollins commented:
@route, I _think_ I have rebased correctly (and fixed a brittle test). Just let me know if there's anything else you need me to do.

And thank you for your kind consideration of my pull request.

vojtad commented:
Hello, is there anything I can do to help to get this merged? We would like to use the Screencast API for debugging flaky tests. Thanks!

route commented:
Thank you!

bradleyscollins commented:
Yes, indeed, my thanks to you as well!
